perfzero/lib/benchmark.py

```diff
@@ -106,108 +102,24 @@ def run_benchmark(self):
-      benchmark_class_name = benchmark_class.rsplit('.', 1)[1]
-
-      tensorflow_profiler = TensorFlowProfiler(
-          self.config.profiler_enabled_time_str, output_dir)
-      process_info_tracker = ProcessInfoTracker(output_dir)
-      process_info = None
-
-      # Setup per-method file logger
-      filehandler = logging.FileHandler(
-          filename=os.path.join(output_dir, 'perfzero.log'), mode='w')
-      filehandler.setFormatter(
-          logging.Formatter('%(asctime)s %(levelname)s: %(message)s'))
-      logging.getLogger().addHandler(filehandler)
-
-      try:
-        class_instance = self._instantiate_benchmark_class(
-            benchmark_class, output_dir, self.config.root_data_dir)
-        # tf.test.Benchmark.report_benchmark() writes results to a file with
-        # path benchmark_result_file_path_prefix + benchmark_method
-        benchmark_result_file_path_prefix = os.path.join(output_dir, 'proto_')
-        os.environ['TEST_REPORT_FILE_PREFIX'] = benchmark_result_file_path_prefix  # pylint: disable=line-too-long
-        benchmark_result_file_path = '{}{}.{}'.format(
-            benchmark_result_file_path_prefix,
-            benchmark_class_name,
-            benchmark_method_name)
-
-        # Start background threads for profiler and system info tracker
-        tensorflow_profiler.start()
-        process_info_tracker.start()
-
-        # Run benchmark method
-        execution_timestamp = time.time()
-        logging.info('Starting benchmark execution: %s', benchmark_method)
-        getattr(class_instance, benchmark_method_name)()
-        logging.info('Stopped benchmark: %s', benchmark_method)
-
-        # Read and build benchmark results
-        raw_benchmark_result = utils.read_benchmark_result(
-            benchmark_result_file_path)
-        # Explicitly overwrite the name to be the full path to benchmark method
-        raw_benchmark_result['name'] = benchmark_method
-      except Exception:  # pylint: disable=broad-except
-        logging.error('Benchmark execution for %s failed due to error:\n %s',
-                      benchmark_method, traceback.format_exc())
-        method_has_exception = True
+      # Run the benchmark method in a separate process so that its memory usage
+      # will not affect the execution of other benchmark method
+      # This is a walkaround before we fix all memory leak issues in TensorFlow
+      queue = multiprocessing.Queue()
+      process = multiprocessing.Process(target=benchmark_method_runner.run,
+                                        args=(benchmark_method,
+                                              site_package_info,
+                                              self.root_output_dir,
+                                              self.config, queue))
+      process.start()
+      process.join()
+      method_has_exception, method_execution_time, succeeded, output_dir = queue.get()  # pylint: disable=line-too-long
+
+      if method_has_exception:
```

Comment:
`has_exception = has_exception or method_has_exception`

Reply:
Thanks for the comment!

Fixed as suggested.